pkg/mcp/resources.go

```diff
@@ -0,0 +1,304 @@
+package mcp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/modelcontextprotocol/go-sdk/mcp"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+	"knative.dev/func/pkg/functions"
+)
+
+// rootHelpResource
+type rootHelpResource struct{}
+
+func (r rootHelpResource) desc() *mcp.Resource {
+	return &mcp.Resource{
+		URI:         "func://help/root",
+		Name:        "Root Help",
+		Description: "--help output of the func command",
+		MIMEType:    "text/plain",
+	}
+}
+
+func (r rootHelpResource) handle(ctx context.Context, request *mcp.ReadResourceRequest, cmdPrefix string, executor Executor) (*mcp.ReadResourceResult, error) {
+	// Parse the command prefix (might be "func" or "kn func")
+	cmdParts := strings.Fields(cmdPrefix)
+	cmdParts = append(cmdParts, "--help")
+
+	content, err := executor.Execute(ctx, "", cmdParts[0], cmdParts[1:]...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &mcp.ReadResourceResult{
+		Contents: []*mcp.ResourceContents{
+			{
+				URI:      "func://help/root",
+				MIMEType: "text/plain",
+				Text:     string(content),
+			},
+		},
+	}, nil
+}
+
+// cmdHelpResource - reusable for command help docs
+type cmdHelpResource struct {
+	cmd []string
+	uri string
+}
+
+func (r cmdHelpResource) desc() *mcp.Resource {
+	return &mcp.Resource{
+		URI:         r.uri,
+		Name:        r.title(),
+		Description: r.description(),
+		MIMEType:    "text/plain",
+	}
+}
+
+// title of the help command resource as a string.
+func (r cmdHelpResource) title() string {
+	titleCaser := cases.Title(language.English)
+	var title string
+	if len(r.cmd) == 1 {
+		title = titleCaser.String(r.cmd[0]) + " Command Help"
+	} else {
+		parts := make([]string, len(r.cmd))
+		for i, part := range r.cmd {
+			parts[i] = titleCaser.String(part)
+		}
+		title = strings.Join(parts, " ") + " Command Help"
+	}
+	return title
+}
+
+// description of the command being resourcified as a string.
+func (r cmdHelpResource) description() string {
+	var desc string
+	if len(r.cmd) == 1 {
+		desc = fmt.Sprintf("--help output of the '%s' command", r.cmd[0])
+	} else {
+		desc = fmt.Sprintf("--help output of the '%s' command", strings.Join(r.cmd, " "))
+	}
+	return desc
+}
```

Comment:
I would use guard pattern where possible, early visible returns and less boilerplate.

```suggestion
func (r cmdHelpResource) description() string {
	if len(r.cmd) == 1 {
		return fmt.Sprintf("--help output of the '%s' command", r.cmd[0])
	} 
	
	return fmt.Sprintf("--help output of the '%s' command", strings.Join(r.cmd, " "))
}
```